### PR TITLE
Product CRUD improvements

### LINKED ISSUE #12326: Refactor simple/grouped product syncing.
Price syncing needs refactoring, possibly similar to how variations and variable products now work under CRUD. Make it automatic.

### PR DESCRIPTION
**Pull request by mikejolley**
This PR fixes the last todo items (that need to be taken care of in this branch, issues are logged for remaining) and makes all unit tests pass. I’ve also scanned for notices and whilst there may be more in testing, all I’ve seen are fixed.

This also closes #12326 which was a todo.

Once merged and @coderkevin completes ajax, we should be able to merge product-crud into master.

### CODE REVIEW

File: includes/api/class-wc-rest-products-controller.php [hunk 1]
```diff
@@ -1180,8 +1178,8 @@ protected function save_product_meta( $product, $request ) {
 			$product->set_parent_id( $request['parent_id'] );
 		}
 
-		// Update parent if grouped so price sorting works and stays in sync with the cheapest child.
-		if ( $product->get_parent_id() > 0 || $product->is_type( 'grouped' ) ) {
+		// Update parent if grouped so price sorting works and stays in sync with the cheapest child. @todo
```
Comment (justinshreve) [inline includes/api/class-wc-rest-products-controller.php]: Isn't this todo complete by the sync code in grouped products? If so - we should remove the whole block

File: includes/class-wc-comments.php [hunk 2]
```diff
@@ -355,7 +355,7 @@ public static function get_average_rating_for_product( $product ) {
 			$average = 0;
 		}
 
-		update_post_meta( $product->get_id(), '_wc_average_rating', true );
+		update_post_meta( $product->get_id(), '_wc_average_rating', $average );
```
Comment (justinshreve) [inline includes/class-wc-comments.php]: These three lines should use CRUD to update.

**Review by emdashcodes — CHANGES_REQUESTED** [review]
Just pointed out two things.
Could you also implement the new delete arguments in the product data store as part of these changes? I already made the change in WC_Data in master, but not in the products store.

**Review by coderkevin — CHANGES_REQUESTED** [review]
In testing some other code, I ran into this in the log: `[15-Nov-2016 23:07:51 UTC] PHP Warning:  implode(): Invalid arguments passed in /srv/www/wordpress-default/wp-content/plugins/woocommerce/includes/wc-attribute-functions.php on line 42[15-Nov-2016 23:07:51 UTC] PHP Warning:  implode(): Invalid arguments passed in /srv/www/wordpress-default/wp-content/plugins/woocommerce/includes/wc-attribute-functions.php on line 42`

It should probably be addressed and fixed before we merge this.

**Comment by justinshreve** [comment]
LGTM 👍

**Pull request merged**